File: src/Entity/Product.php
<?php

namespace App\Entity;

use App\Repository\ProductRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;


#[ORM\Entity(repositoryClass: ProductRepository::class)]
class Product
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Le nom ne peut pas être vide.")]
    #[Assert\Type(type: "string", message: "Le nom doit être une chaîne de caractères.")]
    private ?string $nom = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "La description ne peut pas être vide.")]
    #[Assert\Type(type: "string", message: "La description doit être une chaîne de caractères.")]
    private ?string $description = null;

    #[ORM\Column]
#[Assert\NotBlank(message: "Le prix ne peut pas être vide.")]
#[Assert\Type(type: "numeric", message: "Le prix doit être un nombre.")]
#[Assert\PositiveOrZero(message: "Le prix doit être un nombre positif ou zéro.")]
#[Assert\Regex(
    pattern: "/^\d+(\.\d{1,2})?$/",
    message: "Le prix doit être un nombre valide sans lettres."
)]
private ?float $prix = null;


    #[ORM\Column]
    #[Assert\NotBlank(message: "Le stock ne peut pas être vide.")]
    #[Assert\Type(type: "integer", message: "Le stock doit être un nombre entier.")]
    #[Assert\PositiveOrZero(message: "Le stock doit être un nombre positif ou zéro.")]
    private ?int $stock = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "La catégorie ne peut pas être vide.")]
    private ?string $category = null;
    

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "L'image ne peut pas être vide.")]
    #[Assert\Type(type: "string", message: "L'image doit être une URL sous forme de texte.")]
    private ?string $image = null;

    #[ORM\ManyToOne(targetEntity: User::class, inversedBy: 'products')]
    #[ORM\JoinColumn(nullable: false)]
    #[Assert\NotBlank(message: "L'utilisateur associé ne peut pas être vide.")]
    private ?User $user = null;

    public function getId(): ?int { return $this->id; }

    public function getNom(): ?string { return $this->nom; }
    public function setNom(string $nom): static { $this->nom = $nom; return $this; }

    public function getDescription(): ?string { return $this->description; }
    public function setDescription(string $description): static { $this->description = $description; return $this; }

    public function getPrix(): ?float { return $this->prix; }
    public function setPrix(float $prix): static { $this->prix = $prix; return $this; }

    public function getStock(): ?int { return $this->stock; }
    public function setStock(int $stock): static { $this->stock = $stock; return $this; }

    public function getCategory(): ?string { return $this->category; }
    public function setCategory(string $category): static { $this->category = $category; return $this; }

    public function getImage(): ?string { return $this->image; }
    public function setImage(string $image): static { $this->image = $image; return $this; }

    public function getUser(): ?User { return $this->user; }
    public function setUser(?User $user): static { $this->user = $user; return $this; }
}

File: src/main/java/tn/esprit/models/Product.java
package tn.esprit.models;

import java.util.Date;

public class Product {
    private int id;
    private String nom;
    private String description;
    private double prix;
    private int stock;
    private String category;
    private String image;
    private Date updatedAt;
    private Utilisateur utilisateurs;
    private int utilisateurId;

    // Constructeurs
    public Product() {
    }

    public Product(int id, String nom, String description, double prix, int stock, String category, String image, Date updatedAt) {
        this.id = id;
        this.nom = nom;
        this.description = description;
        this.prix = prix;
        this.stock = stock;
        this.category = category;
        this.image = image;
        this.updatedAt = updatedAt;
    }

    public Product(int id, String nom, String description, double prix, int stock, String category, String image, Date updatedAt, int utilisateurId) {
        this.id = id;
        this.nom = nom;
        this.description = description;
        this.prix = prix;
        this.stock = stock;
        this.category = category;
        this.image = image;
        this.updatedAt = updatedAt;
        this.utilisateurId = utilisateurId;
    }

    // Getters et setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public double getPrix() {
        return prix;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }

    public int getStock() {
        return stock;
    }

    public void setStock(int stock) {
        this.stock = stock;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Date updatedAt) {
        this.updatedAt = updatedAt;
    }

    public Utilisateur getUtilisateurs() {
        return utilisateurs;
    }

    public void setUtilisateur(Utilisateur utilisateurs) {
        this.utilisateurs = utilisateurs;
    }

    public int getUtilisateurId() {
        return utilisateurId;
    }

    public void setUtilisateurId(int utilisateurId) {
        this.utilisateurId = utilisateurId;
    }

    @Override
    public String toString() {
        return "Product{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", description='" + description + '\'' +
                ", prix=" + prix +
                ", stock=" + stock +
                ", category='" + category + '\'' +
                ", image='" + image + '\'' +
                ", updatedAt=" + updatedAt +
                ", utilisateurId=" + utilisateurId +
                '}';
    }
}